core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java

```diff
@@ -179,7 +179,8 @@ void deleteInternal(@PathParam("workspaceId") Long workspaceId, @PathParam("name
     @ApiOperation(value = UPDATE_BY_NAME_IN_WORKSPACE, produces = MediaType.APPLICATION_JSON, notes = Notes.STACK_NOTES,
             nickname = "updateNameCrnAndType")
     void updateNameAndCrn(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name,
-            @QueryParam("initiatorUserCrn") String initiatorUserCrn, @QueryParam("newName") String newName, @QueryParam("newCrn") String newCrn);
+            @QueryParam("initiatorUserCrn") String initiatorUserCrn, @QueryParam("newName") String newName, @QueryParam("newCrn") String newCrn,
+            @QueryParam("detached") @DefaultValue("false") boolean detached);
```

Comment:
There is no need for a detached variable.

Whenever updateNameAndCrn is invoked, the originalCrn field can be initialized to retain the old name.